bokehjs/src/coffee/models/axes/axis.coffee

```diff
@@ -314,54 +314,51 @@ class Axis extends GuideRenderer.Model
       x_range_name:   [ p.String,   'default' ]
       y_range_name:   [ p.String,   'default' ]
       axis_label:     [ p.String,   ''        ]
+      axis_label_standoff:     [ p.Int,         5 ]
+      major_label_standoff:    [ p.Int,         5 ]
+      major_label_orientation: [ p.Any, "horizontal" ] # TODO: p.Orientation | p.Number
       major_tick_in:  [ p.Number,   2         ]
       major_tick_out: [ p.Number,   6         ]
       minor_tick_in:  [ p.Number,   0         ]
       minor_tick_out: [ p.Number,   4         ]
-    }
-
-  defaults: ->
-    return _.extend {}, super(), {
-      # overrides
-      axis_line_color: 'black'
+  }
 
-      major_tick_line_color: 'black'
-      minor_tick_line_color: 'black'
+  @override {
+    axis_line_color: 'black'
 
-      major_label_standoff: 5
-      major_label_orientation: "horizontal"
-      major_label_text_font_size: "10pt"
-      major_label_text_align: "center"
-      major_label_text_baseline: "alphabetic"
+    major_tick_line_color: 'black'
+    minor_tick_line_color: 'black'
 
-      axis_label_standoff: 5
-      axis_label_text_font_size: "16pt"
-      axis_label_text_align: "center"
-      axis_label_text_baseline: "alphabetic"
+    major_label_text_font_size: "10pt"
+    major_label_text_align: "center"
+    major_label_text_baseline: "alphabetic"
 
-      # internal
-    }
+    axis_label_text_font_size: "16pt"
+    axis_label_text_align: "center"
+    axis_label_text_baseline: "alphabetic"
+  }
 
-  nonserializable_attribute_names: () ->
-    super().concat(['layout_location'])
+  @internal {
+    layout_location: [ p.Any ]
+  }
 
   initialize: (attrs, options)->
     super(attrs, options)
 
-    @register_property('computed_bounds', @_computed_bounds, false)
+    @define_computed_property('computed_bounds', @_computed_bounds, false)
     @add_dependencies('computed_bounds', this, ['bounds'])
     @add_dependencies('computed_bounds', @get('plot'), ['x_range', 'y_range'])
 
-    @register_property('rule_coords', @_rule_coords, false)
+    @define_computed_property('rule_coords', @_rule_coords, false)
     @add_dependencies('rule_coords', this, ['computed_bounds', 'side'])
 
-    @register_property('tick_coords', @_tick_coords, false)
+    @define_computed_property('tick_coords', @_tick_coords, false)
     @add_dependencies('tick_coords', this, ['computed_bounds', 'layout_location'])
 
-    @register_property('ranges', @_ranges, true)
-    @register_property('normals', (() -> @_normals), true)
-    @register_property('dimension', (() -> @_dim), true)
-    @register_property('offsets', @_offsets, true)
+    @define_computed_property('ranges', @_ranges, true)
+    @define_computed_property('normals', (() -> @_normals), true)
+    @define_computed_property('dimension', (() -> @_dim), true)
+    @define_computed_property('offsets', @_offsets, true)
```

Comment:
Thanks for the better name, FWIW I would like to eventually get rid of computed properties entirely.